Ignore internal tables in MoveTables and Reshard Workflows

mattlord opened the pull request:
## Description
We need to ignore internal operation tables (e.g. for OnlineDDL processing) in `MoveTables` and `Reshard` workflows. We need to do this both when processing the `-tables` command-line flags and when processing `SchemaDefinition` structures that we get from the database.

I also got rid of the logging messages for when these internal tables are seen and ignored/skipped as I don't think that they are helpful to the end user.

<details>
<summary>You can see a simple Reshard demonstration and test here</summary>

These are using the [docker_local container](https://vitess.io/docs/13.0/get-started/local-docker/).

### Before (main):
```
$ make docker_local && ./docker/local/run.sh
./201_customer_tablets.sh ; ./202_move_tables.sh ; sleep 10; ./203_switch_reads.sh ; ./204_switch_writes.sh ; ./205_clean_commerce.sh ; ./301_customer_sharded.sh ; ./302_new_shards.sh; sleep 5; mysql customer -e "create table _vt_PURGE_1f9194b43b2011eb8a0104ed332e05c2_20201210194431(id int, val varbinary(128), primary key(id))"; ./303_reshard.sh ; sleep 5; mysql customer/-80 -e "show tables"
...
+-----------------------------------------------------------+
| Tables_in_vt_customer                                     |
+-----------------------------------------------------------+
| _vt_PURGE_1f9194b43b2011eb8a0104ed332e05c2_20201210194431 |
| corder                                                    |
| customer                                                  |
+-----------------------------------------------------------+
```

### After (this branch):
```
$ make docker_local && ./docker/local/run.sh
./201_customer_tablets.sh ; ./202_move_tables.sh ; sleep 10; ./203_switch_reads.sh ; ./204_switch_writes.sh ; ./205_clean_commerce.sh ; ./301_customer_sharded.sh ; ./302_new_shards.sh; sleep 5; mysql customer -e "create table _vt_PURGE_1f9194b43b2011eb8a0104ed332e05c2_20201210194431(id int, val varbinary(128), primary key(id))"; ./303_reshard.sh ; sleep 5; mysql customer/-80 -e "show tables"
...
+-----------------------+
| Tables_in_vt_customer |
+-----------------------+
| corder                |
| customer              |
+-----------------------+
```
</details>

## Related Issue(s)
Follow-up to: https://github.com/vitessio/vitess/pull/8992
Fixes: https://github.com/vitessio/vitess/issues/9577

## Checklist
- [x] Should this PR be backported? No
- [x] Tests were added
- [x] Documentation is not required (users should not even be aware of these tables)

mattlord commented:
@shlomi-noach could you please offer some insights/input here if you have time? You may be aware of potential problems in eliding the internal tables at the `mysqlctl.GetSchema` layer. If so, I'll have to find more locations to explicitly ignore these internal tables (Reshard still fails w/o that and will have to find an alternative fix).

mattlord commented:
> @shlomi-noach could you please offer some insights/input here if you have time? You may be aware of potential problems in eliding the internal tables at the `mysqlctl.GetSchema` layer. If so, I'll have to find more locations to explicitly ignore these internal tables (Reshard still fails w/o that and will have to find an alternative fix).

The failed tests seem to be telling me it is a problem. :-) I'll abandon that approach, at least for now.

rohit-nayak-ps commented on go/vt/mysqlctl/tmutils/schema.go:
```diff
@@ -273,9 +275,7 @@ func DiffSchema(leftName string, left *tabletmanagerdatapb.SchemaDefinition, rig
 		// extra table on the left side
 		if left.TableDefinitions[leftIndex].Name < right.TableDefinitions[rightIndex].Name {
-			if schema.IsInternalOperationTableName(left.TableDefinitions[leftIndex].Name) {
-				log.Infof("found internal table %v, skipping in schema diff", left.TableDefinitions[leftIndex].Name)
```
The reason we had added these logs were for visibility in case some table mysteriously failed to get resharded and the user could identify why from the logs. But removing it should be fine.

mattlord replied:
Yeah, that was my thinking as well too (I added these specific log messages here). But it ends up as a lot of log messages in total and makes it harder to find the signal in the noise. I feel comfortable removing them as I double checked the REGEXPs and they are very precise:
  - https://github.com/vitessio/vitess/blob/a033d76e25a1ce5332b3956f8db9801f07bccb81/go/vt/schema/tablegc.go#L47
  - https://github.com/vitessio/vitess/blob/a033d76e25a1ce5332b3956f8db9801f07bccb81/go/vt/schema/online_ddl.go#L39

rohit-nayak-ps commented on go/vt/wrangler/materializer.go:
```diff
@@ -93,6 +93,26 @@ func (wr *Wrangler) addTablesToVSchema(ctx context.Context, sourceKeyspace strin
+	// structures built from the GetSchema database related API calls. In this
+	// case, however, the table list comes from the user via the -tables flag
+	// so we need to filter out internal table names here in case a user has
+	// explcility specified some.
```
typo explcility=>explicitly. Maybe fix later, no need for CI to run again for this!

rohit-nayak-ps reviewed (approved):
lgtm

shlomi-noach commented on go/test/endtoend/vreplication/helper.go:
```diff
@@ -92,6 +93,22 @@ func waitForQueryToExecute(t *testing.T, conn *mysql.Conn, database string, quer
+	require.NotNil(t, qr)
+	require.NotNil(t, qr.Rows)
+	found := false
+	for _, row := range qr.Rows {
```
To gain more visibility into failure, I'd suggest:
```go
   tableName := row[0].ToString()
   assert.False(t, schema.IsInternalOperationTableName(tableName), "found internal table %s", tableName)
```

then, remove the block with `t.Logf` and the `require...` that follows.

mattlord replied:
Good call. I was being noisy during testing to see what was happening, but don't need that in ever CI run going forward.

mattlord commented:
Thank you, @shlomi-noach and @rohit-nayak-ps! I've addressed those comments here: https://github.com/vitessio/vitess/pull/9578/commits/1101339d4267788583ec9222204867eccfc453cc
